### ISSUE TITLE
Issue with nested templates

### ISSUE THREAD

**Issue by vtatarin** [hairyhenderson/gomplate#1600]
Hello! First of all, thank you for such a great tool.

Looks like we found an issue related to the nested templates configuration or we are not fully understanding how it should behave. You can find a detailed report below:

**Context**
We would like to move a part of the templating logic into a separate file (similar to how Helm templating mechanism has a `_helpers.tpl` file). Also, we are expecting full configuration of gomplate to be in a file (`.gomplate.yaml`). 

**AR**
```
$ cat .gomplate.yaml
templates:
  h:
    url: ./helpers.tmpl

in: '{{ template "h" "world" }}'
```

$ gomplate -V --config=.gomplate.yaml
$ echo $?
0

After we add the `templates` section even to a config file with `inputDir`/`outputMap`, `gomplate` stops any rendering.
**ER**
The output should be similar to the one where configuration of gomplate is passed as cli args:
```
$ gomplate -V -t h=helpers.tmpl  -i '{{ template "h" "world" }}'
13:37:53 DBG using config file cfgFile=.gomplate.yaml
13:37:53 DBG starting gomplate
13:37:53 DBG config is:
---
templates:
  h:
    url: file:///Users/path/to/helpers.tmpl
in: '{{ templ...'
 build= version=3.11.3
Hello world!
13:37:53 DBG completed rendering duration=0.001560609 errors=0 templatesRendered=1
```

Content of `helpers.tmpl`:
```
Hello {{ . }}!
```

Additional info:
```
$ gomplate -v
gomplate version 3.11.3
$ uname -a
Darwin MacBook-Pro.local 22.1.0 Darwin Kernel Version 22.1.0: Sun Oct  9 20:14:54 PDT 2022; root:xnu-8792.41.9~2/RELEASE_X86_64 x86_64
```

Thank you in advance!

**Comment by hairyhenderson** [comment]
Hi @bugaga1100, thanks for filing this.

It looks like this works fine with the current code that's in `main`, but I can reproduce the failure with the 3.11.3 binary.

I'll take a look when I have some free time to see if I can find if/where the bug was fixed.

**Comment by bugaga1100** [comment]
@hairyhenderson thank you for the fast reply. Do you plan to publish a new release soon (it has already been a while since September)?

**Comment by hairyhenderson** [comment]
I have limited time to work on gomplate these days, unfortunately, and so releases are sporadic and best-effort.

**Comment by bugaga1100** [comment]
It would be great to publish a new release to get this issue fixed since it's already in the main branch. Thank you for your time!

**Comment by hairyhenderson** [comment]
The main blocker is there's a _lot_ in `main`, including some breaking changes that will necessitate a v4.0.0 release. I'm working towards it (slowly), but it'll take more time.

I'm still going to try to see if I can narrow down what exactly the bug is - maybe it's easy enough to backport for a patch release. But again, that'll take time 😉

**Comment by RafPe** [comment]
@hairyhenderson As this one still persists - is there some help we could offer from our side under your guidance ?

**Comment by hairyhenderson** [comment]
Hi @RafPe - one thing to do would be to determine what fixed the bug between `v3.11.3` and `main` - that would involve a bunch of git bisecting.

**Comment by RafPe** [comment]
@hairyhenderson is there a test written for this case that I could run in order to help me narrow down the piece of code responsible for the fix ?
